sprint0/sprint0/HUD/HUDManager.cs

```diff
@@ -52,6 +51,12 @@ public void LoadHUD()
             populateHUDInventory.PopulateInventoryHUD();
         }
         public bool HasItem(PlayerItems item) => pauseInventory.HasItem(item);
+        public bool HasKeys() => populateHUDInventory.GetNum(PlayerItems.Key) > 0;
+        public void DecrementKey()
+        {
+            if (!HasItem(PlayerItems.MagicalKey) && HasKeys())
```

Comment:
what if link has both Magical Keys and normal keys? we still want to decrement key if it is used to open a door